src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1172,53 +1155,41 @@ private void ParseModifiers(SyntaxListBuilder tokens, bool forAccessors)
                 switch (newMod)
                 {
                     case DeclarationModifiers.Partial:
-                        var nextToken = PeekToken(1);
-                        var isPartialType = this.IsPartialType();
-                        var isPartialMember = this.IsPartialMember();
-                        if (isPartialType || isPartialMember)
                         {
-                            // Standard legal cases.
+                            var flags = ShouldPartialBeTreatedAsModifier();
+                            if (flags == PartialModFlags.NotModifier)
+                            {
+                                return;
+                            }
+
                             modTok = ConvertToKeyword(this.EatToken());
+                            if (flags == PartialModFlags.TreatAsModifier)
+                            {
+                                // Error reported in binding
+                                break;
```

Comment:
sorry, it's not clear to me what's going on.  Just from reading, here's what i see.  You got a 'partial' token.  You saw that it *should* be treated as a 'partial' modifier.  But then you break... saying "error reported in binding".  It's not clear ot me in this code why this is an error path.  It seems like the "correct" path.  So something isn't gelling.  Needs better comment if this really is correct here.

Reply:
It's confusing just because we return early. TreatAsModifier means that it is actually a modifier but we don't have a partial type or partial member. Will add a comment.

Reply:
But why return early.  The function is called "ParseModifiers".  You had a modifier.  Indeed, it says "TreatAsModifier" :)   So... even if it's not a type or member... why not still consume this modifier.  I don't get why bailing is the write thing here...

Reply:
we consume it just before this if. I factored out that line for all the remaining "success" cases. Probably I shouldn't have.